projects/manage-supervision-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/service/SentenceService.kt

```diff
@@ -116,6 +116,50 @@ class SentenceService(
         return null
     }
 
+    fun getUnpaidWorkTime(disposalId: Long): String? {
+        val totalHoursOrdered = requirementRepository.sumTotalUnpaidWorkHoursByDisposal(disposalId)
+
+        if (totalHoursOrdered == 0L) {
+            return null
+        }
+
+        val totalMessage = when (totalHoursOrdered) {
+            1L -> "(of $totalHoursOrdered hour)"
+            else -> "(of $totalHoursOrdered hours)"
+        }
+
+        val durationInMinutes: Long = upwAppointmentRepository.calculateUnpaidTimeWorked(disposalId)
+        durationInMinutes.let {
+            val duration = Duration.ofMinutes(it)
+            val completed = "completed"
+            val hour = "hour"
+            val hours = "hours"
+            val minuteCompleted = "minute $completed"
+            val minutesCompleted = "minutes $completed"
+
+            val unpaidWorkHours = when (duration.toHoursPart()) {
+                0 -> null
+
+                1 -> "${duration.toHoursPart()} $hour"
+
+                else -> "${duration.toHoursPart()} $hours"
+            }
+
+            val unpaidWorkMinutes = when (duration.toMinutesPart()) {
+                0 -> null
+
+                1 -> "${duration.toMinutesPart()} $minuteCompleted"
+
+                else -> "${duration.toMinutesPart()} $minutesCompleted"
+            }
+
+            unpaidWorkHours ?: unpaidWorkMinutes ?: return "no time completed (of 1 hour)"
+            unpaidWorkHours ?: return "$unpaidWorkMinutes $totalMessage"
+            unpaidWorkMinutes ?: return "$unpaidWorkHours $completed $totalMessage"
+            return "$unpaidWorkHours and $unpaidWorkMinutes $totalMessage"
+        }
+    }
```

Comment:
"(of 1 hour)" is hard-coded on line 156, also you might be able to simplify this with the Kotlin duration extension - e.g. something like:
```suggestion
    fun getUnpaidWorkTime(disposalId: Long): String {
        val hoursOrdered = requirementRepository.sumTotalUnpaidWorkHoursByDisposal(disposalId)
        val totalMessage = hoursOrdered
            ?.let { " (of $hoursOrdered hour${if (hoursOrdered != 1L) "s" else ""})"  } ?: ""

        val minutesCredited = upwAppointmentRepository.calculateUnpaidTimeWorked(eventId, eventNumber)
        val creditedMessage = Duration.ofMinutes(minutesCredited).toKotlinDuration()
            .toComponents { hours, minutes, _, _ -> 
                when {
                    hours == 0L -> "$minutes minute${if (minutes != 1) "s" else ""} completed"
                    minutes == 0 -> "$hours hour${if (hours != 1L) "s" else ""} completed"
                    else -> "$hours hour${if (hours != 1L) "s" else ""} $minutes minute${if (minutes != 1) "s" else ""} completed"
                }
            }
        return "$creditedMessage$totalMessage"
    }
```

Reply:
You can play around with a few test cases in the playground here: https://pl.kotl.in/Don_1E9j7

Reply:
updated